Dockerfile

```diff
@@ -1,14 +1,31 @@
-FROM python:3-alpine
-RUN apk update && apk add git openssh-client
+FROM python:3-alpine as base
 
-COPY . /app
 WORKDIR /app
 
-RUN pip install -e .
+FROM base as builder
 
+ENV PIP_DEFAULT_TIMEOUT=100 \
+    PIP_DISABLE_PIP_VERSION_CHECK=1 \
+    PIP_NO_CACHE_DIR=1 \
+    POETRY_VERSION=1.0.5
+
+RUN apk add --no-cache gcc libffi-dev musl-dev openssl-dev
+RUN pip --no-cache-dir install "poetry==$POETRY_VERSION"
+RUN python -m venv /venv
+
+COPY pyproject.toml poetry.lock ./
+RUN poetry export -f requirements.txt | /venv/bin/pip install -r /dev/stdin
+
+COPY . .
+RUN poetry build && /venv/bin/pip install dist/*.whl
```

Comment:
Curious why we are going this route, rather than just using `poetry install` and `poetry build`?

Reply:
The idea here is that we ship a final image with none of the build artifacts; only the finally installed product. If you notice, poetry is not installed into the venv, and so doesn't actually get copied over to the final image. Same with the `-dev` packages installed on line 12. The main goal here is to provide as compact and clean an image as we can.

So with all of that in mind, the easiest way to get Poetry to use the Python interpreter in the venv, is to do this export and pip install.

Of course, all this is adapted from a number of examples I've found online. So I'd be happy if somebody is able to help build a better way!